Library/Homebrew/cask/lib/hbc/cli/search.rb

```diff
@@ -2,8 +2,12 @@ module Hbc
   class CLI
     class Search < AbstractCommand
       def run
-        results = self.class.search(*args)
-        self.class.render_results(*results)
+        if args.empty?
+          puts Formatter.columns(CLI.nice_listing(Hbc.all_tokens))
```

Comment:
The whole point of this change is that `brew cask search` (without arguments) _should not_ perform online search, same way is `brew search` does (see [here](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cmd/search.rb#L26-L27)). 
You can still do `brew cask search ""` which will do a normal search returning all casks.